src/v/transform/logging/log_manager.cc

```diff
@@ -0,0 +1,297 @@
+      : _as(as)
+      , _client(c)
+      , _interval_ms(std::move(fi))
+      , _jitter(_interval_ms(), jitter_amt) {
+        _interval_ms.watch([this]() {
+            _jitter = simple_time_jitter<ClockType>{_interval_ms(), jitter_amt};
+        });
+    }
+
+    template<typename BuffersT>
+    ss::future<> start(ss::gate& gate, BuffersT* bufs) {
+        vassert(
+          bufs != nullptr, "Pointer to transform log buffers must not be null");
+        ssx::spawn_with_gate(
+          gate, [this, bufs]() -> ss::future<> { return flush_loop(bufs); });
+        return ss::now();
+    }
+
+    void wakeup() { _wakeup_signal.signal(); }
+
+    ss::future<> stop() {
+        _wakeup_signal.broken();
+        return ss::now();
+    }
+
+    template<typename BuffersT>
+    ss::future<> flush_loop(BuffersT* bufs) {
+        while (!_as->abort_requested()) {
+            try {
+                // the duration overload passes now() + dur to the timepoint
+                // overload, but the now() calculation is tied to an underlying
+                // clock type, so it doesn't work with ss::manual_clock. the
+                // timepoint overload template has a clocktype parameter, so we
+                // use that one to get the behavior we want for testing
+                co_await _wakeup_signal.wait<ClockType>(
+                  ClockType::now() + _jitter.base_duration());
+
+            } catch (const ss::broken_condition_variable&) {
+                break;
+            } catch (const ss::condition_variable_timed_out&) {
+            }
+            co_await flush(bufs);
+        }
+        co_return;
+    }
+
+    template<typename BuffersT>
+    ss::future<> flush(BuffersT* bufs) {
+        size_t n_events = 0;
+        absl::flat_hash_map<model::partition_id, io::json_batches> batches{};
+        BuffersT local_bufs;
+        for (auto& [name, buf] : *bufs) {
+            if (buf.empty()) {
+                continue;
+            }
+            n_events += buf.size();
+            local_bufs.emplace(name, std::exchange(buf, {}));
+        }
+
+        co_await ss::max_concurrent_for_each(
+          local_bufs, MAX_CONCURRENCY, [this, &batches](auto& pr) {
+              auto& [name, events] = pr;
+              model::transform_name_view nv{name};
+              auto pid = _client->compute_output_partition(nv);
+
+              return do_serialize_log_events(nv, std::move(events))
+                .then([pid, &batches](auto batch) -> ss::future<> {
+                    batches[pid].emplace_back(std::move(batch));
+                    return ss::now();
+                });
+          });
+
+        co_await ss::max_concurrent_for_each(
+          batches, MAX_CONCURRENCY, [this](auto& pr) mutable -> ss::future<> {
+              auto& [pid, evs] = pr;
+              return do_flush(pid, std::move(evs));
+          });
+
+        vlog(tlg_log.trace, "Processed {} log events", n_events);
+        co_return;
```

Comment:
nit: `co_return` isn't needed.

Reply:
yeah, cruft. i had a status code there, then removed it, added it back, removed it.